tests/functional/ca/issue-13247.sh

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env bash
+
+# https://github.com/NixOS/nix/issues/13247
+
+export NIX_TESTS_CA_BY_DEFAULT=1
+
+source common.sh
+
+clearStoreIfPossible
+
+set -x
+
+# Build derivation (both outputs)
+nix build -f issue-13247.nix --json a a-prime use-a-more-outputs --no-link > "$TEST_ROOT"/a.json
+
+cache="file://$TEST_ROOT/cache"
+
+# Copy all outputs and realisations to cache
+declare -a drvs
+for d in "$NIX_STORE_DIR"/*-issue-13247-a.drv "$NIX_STORE_DIR"/*-use-a-more-outputs.drv; do
+    drvs+=("$d" "$d"^*)
+done
+nix copy --to "$cache" "${drvs[@]}"
+
+function delete () {
+    # Delete local copy
+    # shellcheck disable=SC2046
+    nix-store --delete \
+        $(jq -r <"$TEST_ROOT"/a.json '.[] | .drvPath, .outputs.[]') \
+        "$NIX_STORE_DIR"/*-issue-13247-a.drv \
+        "$NIX_STORE_DIR"/*-use-a-more-outputs.drv
+
+    [[ ! -e "$(jq -r <"$TEST_ROOT"/a.json '.[0].outputs.out')" ]]
+    [[ ! -e "$(jq -r <"$TEST_ROOT"/a.json '.[1].outputs.out')" ]]
+    [[ ! -e "$(jq -r <"$TEST_ROOT"/a.json '.[2].outputs.first')" ]]
+    [[ ! -e "$(jq -r <"$TEST_ROOT"/a.json '.[2].outputs.second')" ]]
+}
+
+delete
+
+buildViaSubstitute () {
+    nix build -f issue-13247.nix "$1" --no-link --max-jobs 0 --substituters "$cache" --no-require-sigs --offline --substitute
```

Comment:
I had to add `--substitute` to not make the lack of network access disable substituters. (I added `--offline` too simply to reproduce the issue in dev builds.)

I am curious why the other substitution tests are not affected by this.

Also, no network should not disable substitutes that don't use the network!